packages/frontend/core/src/components/pure/workspace-slider-bar/favorite/favorite-list.tsx

```diff
@@ -15,11 +17,18 @@ export const FavoriteList = ({
   docCollection: workspace,
 }: FavoriteListProps) => {
   const metas = useBlockSuiteDocMeta(workspace);
+  const favAdapter = useService(FavoriteItemsAdapter);
   const dropItemId = getDropItemId('favorites');
 
+  const favourites = useLiveData(favAdapter.favorites$);
```

Comment:
You can derive one livedata from another using `map`.

This can prevent the filtered parts causing react to rerender.

```
favAdapter.favorites$.map((favourites) => {
    return favourites.filter(fav => {
      const meta = metas.find(m => m.id === fav.id);
      return meta && !meta.trash;
    })
})
```

Reply:
I did not do this because of the same reason of performance concerns about sub/unsub on rerender.
What do you think?

Reply:
sub unsub should be fast. Much faster than react rerender